cheroot/server.py

```diff
@@ -1353,7 +1351,25 @@ def close(self):
         self.rfile.close()
 
         if not self.linger:
-            self._close_kernel_socket()
+            try:
```

Comment:
I believe that try/except should only wrap the shutdown call, not the logic of retrieving the proper method.